ls1intum/Artemis
# Enhancement/compass improvements

### Checklist

### Motivation and Context
1. Enable multi node setup for compass
2. Remove deprecated methods and unnecessary interfaces

### Description
1. Uses hazelcast for keeping compass data
2. Removes deprecated assessment methods from diagram
3. Move assessment methods to assessment controller
4. Remove assessment index that only holds a map and its method
5. Remove calculation engine interface
6. Stop assessing on modeling submission but move that functionality to adding assessment

### Steps for Testing
1. Create modeling exercise with class diagram and make sure compass is enabled
2. Login as several users and submit submissions
3. Assess manually a submission
4. Your assessment must be suggested for similar elements of other submissions
5. Modify suggested assessment and check if the longer text is suggested for the next similar element
6. Submit a new assessment with a different user and check if artemis suggests assessment for that submission as well
